### ISSUE TITLE
Quarkus OIDC Multitenancy won't work in DEV mode

### ISSUE THREAD

**Issue by anamak9** [quarkusio/quarkus#37439]
### Describe the bug

The dev mode is not picking the tenant specific OIDC configuration dynamically (request scope).

### Expected behavior

If the request is coming for a default config (determined thru a URL path or a header) then default OIDC config should be used, likewise same should happen for a tenant specific requests. 

### Actual behavior

While the app is booting up, the default OIDC is successfully registered but afterwards it never changes on a client request basis (as should be expected from a given multi-tenancy configurations).

Logs:

[INFO] Changes detected - recompiling the module! :dependency
Listening for transport dt_socket at address: 8005
2023-11-30 12:49:54,017 INFO  [io.qua.oid.dep.dev.OidcDevUIProcessor] (build-17) **OIDC Dev Console: discovering the provider metadata at https://cognito-idp.us-west-1.amazonaws.com/{user pool id}/.well-known/openid-configuration**
2023-11-30 12:49:54,359 INFO  [io.qua.ama.lam.run.MockEventServer] (build-45) Mock Lambda Event Server Started
__  ____  __  _____   ___  __ ____  ______
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
2023-11-30 12:50:01,134 WARN  [io.qua.oid.run.TenantConfigContext] (vert.x-eventloop-thread-1) Secret key for encrypting tokens in a session cookie is missing, auto-generating it

2023-11-30 12:50:01,341 WARN  [io.qua.oid.run.TenantConfigContext] (vert.x-eventloop-thread-1) Secret key for encrypting tokens in a session cookie is missing, auto-generating it
2023-11-30 12:50:01,628 INFO  [io.qua.ama.lam.run.AbstractLambdaPollLoop] (Lambda Thread (DEVELOPMENT)) Listening on: http://localhost:9090/_lambda_/2018-06-01/runtime/invocation/next
2023-11-30 12:50:01,628 INFO  [io.quarkus] (Quarkus Main Thread) my-app 1.0 on JVM (powered by Quarkus 3.5.0) started in 13.727s.
2023-11-30 12:50:01,636 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2023-11-30 12:50:01,639 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [amazon-lambda, cache, cdi, hibernate-validator, mongodb-client, mongodb-panache, narayana-jta, oidc, resteasy-reactive, resteasy-reactive-jackson, security, smallrye-context-propagation, smallrye-jwt, smallrye-openapi, swagger-ui, vertx]
**2023-11-30 12:50:40,143 INFO  [com.sli.res.CustomTenantResolver] (vert.x-eventloop-thread-2) User request role: admin**

### How to Reproduce?

- Add following to your application.properties,

```
# Default (regular-user) Tenant Configuration
quarkus.oidc.auth-server-url=<aws cognito user pool url>
quarkus.oidc.client-id=<aws cognito userpool id>
quarkus.oidc.application-type=web-app
quarkus.oidc.token-path=/oauth2/token

# Tenant-a  Configuration
quarkus.oidc.tenant-a.auth-server-url=<aws cognito user pool url for tenant-a>
quarkus.oidc.tenant-a.client-id=<aws cognito userpool id for tenant-a>
quarkus.oidc.tenant-a.application-type=web-app
quarkus.oidc.tenant-a.token-path=/oauth2/token
```

- Then create a resolver class for it,

`public class CustomTenantResolver implements TenantResolver { ... }`

 In other words, follow the steps as mentioned on the below guide,

https://quarkus.io/guides/security-openid-connect-multitenancy

- Finally make two kinds of HTTP (GET or POST or whatever) requests attached with a valid JWT access token, one intended for to be handled by default OIDC config and the other for 'tenant-a' specific config.
